toolchain/check/subst.cpp

```diff
@@ -63,7 +63,9 @@ static auto PushOperand(Context& context, Worklist& worklist,
       worklist.Push(static_cast<SemIR::InstId>(arg));
       break;
     case SemIR::IdKind::For<SemIR::TypeId>:
-      worklist.Push(context.types().GetInstId(static_cast<SemIR::TypeId>(arg)));
+      if (auto type_id = static_cast<SemIR::TypeId>(arg); type_id.is_valid()) {
```

Comment:
Why do you use static_cast here instead of `SemIR::TypeId(arg)`? I grant the code was doing this previously, but it seems inconsistent (e.g., in ValueStore, we have `IdT id = IdT(values_.size());`, not `static_cast<IdT>(values_.size())`). I'm actually a little surprised this works with the explicit constructors.

I see you add `SemIR::TypeId(type_id)` on line 157, and `static_cast<SemIR::TypeId>(arg)` on line 106. I do think we should be consistent on this.

Similar elsewhere in this file -- a quick scan suggests this may be the only file using static_cast for Id types. If you prefer to do cleanup separately, that's fine, though I'm also fine if you were to just remove the static_casts within this PR -- it looks like there's 8 pre-existing (I don't count the `static_cast<int>`).